testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,363 @@
+/*
+Copyright © 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package transfigure
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/spf13/cobra"
+)
+
+const Branch = "transfigure-branch"
+
+var (
+	dryRun         bool
+	gitEmail       string
+	githubToken    string
+	gitUser        string
+	prowConfig     string
+	prowJobConfig  string
+	prTitle        string
+	remoteForkRepo string
+	repoSubdir     string
+	tokenContents  string
+	testgridDir    string
+	testgridYaml   string
+	workingDir     string
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
```

Comment:
I'm afraid that using cobra is an overkill in this case, since there is no subcommand for this tool, use standard flag parsing as https://github.com/kubernetes/test-infra/blob/6d943af9d0a0bc93da838ba0ab19414e0370306c/gencred/cmd/gencred/main.go#L63 should be suffice

Reply:
I really like Cobra for things like automatic help generation and would rather not write those things myself. Is it ok to leave as is?